[Maps] Support by value saved objects.

nreese opened the pull request:
This PR updates map embeddable and map application to support saved objects by value, allowing users to store map saved objects directly in dashboard without needed to create and maintain a separate saved object.

Testing instructions
1) Add `dashboard.allowByValueEmbeddables: true` in kibana.dev.yml
2) Create new dashboard
3) Use `create new` to add a panel and select maps.
4) Create your map and save with `Save return`
5) save your dashboard. Inspect dashboard saved object. Notice how map saved object is stored directly in dashboard.

elasticmachine commented:
Pinging @elastic/kibana-gis (Team:Geo)

nreese commented on x-pack/plugins/maps/public/routing/routes/map_app/map_app.tsx:
```diff
@@ -86,7 +87,6 @@ interface Props {
   refreshConfig: MapRefreshConfig;
   setRefreshConfig: (refreshConfig: MapRefreshConfig) => void;
   isSaveDisabled: boolean;
-  clearUi: () => void;
```
clearUi no longer needed since each savedMap has its own reduce store. Since a store is not being shared, there is nothing to clean up

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85621/)
* Commit: b7bf3bed23364d4f5059709392cf03cc6d49ce8b
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85621/execution/node/381/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85621/","number":"85621","commit":"b7bf3bed23364d4f5059709392cf03cc6d49ce8b"}]}
PIPELINE-->

nreese commented on x-pack/plugins/maps/public/routing/store_operations.ts:
```diff
@@ -1,11 +0,0 @@
-
-import { createMapStore } from '../reducers/store';
-
-const store = createMapStore();
```
Removed, store is no longer a singleton. Each map has its own store.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85632/)
* Commit: 37e3372fa40e1eb656233c7011c6f65b7d48658e
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85632/execution/node/379/log)

### History
* :broken_heart: [Build #85621](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85621/) failed b7bf3bed23364d4f5059709392cf03cc6d49ce8b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85621/","number":"85621","commit":"b7bf3bed23364d4f5059709392cf03cc6d49ce8b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85632/","number":"85632","commit":"37e3372fa40e1eb656233c7011c6f65b7d48658e"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85687/)
* Commit: 5e21aecfc057655a022b2ca8c92aea176164156f
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85687/execution/node/381/log)

### History
* :broken_heart: [Build #85632](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85632/) failed 37e3372fa40e1eb656233c7011c6f65b7d48658e
* :broken_heart: [Build #85621](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85621/) failed b7bf3bed23364d4f5059709392cf03cc6d49ce8b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85621/","number":"85621","commit":"b7bf3bed23364d4f5059709392cf03cc6d49ce8b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85632/","number":"85632","commit":"37e3372fa40e1eb656233c7011c6f65b7d48658e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85687/","number":"85687","commit":"5e21aecfc057655a022b2ca8c92aea176164156f"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85786/)
* Commit: 87126ad8dc4fb7fe67a5541bb7dd38b27b7f38bd
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85786/execution/node/371/log)

### History
* :broken_heart: [Build #85687](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85687/) failed 5e21aecfc057655a022b2ca8c92aea176164156f
* :broken_heart: [Build #85632](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85632/) failed 37e3372fa40e1eb656233c7011c6f65b7d48658e
* :broken_heart: [Build #85621](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85621/) failed b7bf3bed23364d4f5059709392cf03cc6d49ce8b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85621/","number":"85621","commit":"b7bf3bed23364d4f5059709392cf03cc6d49ce8b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85632/","number":"85632","commit":"37e3372fa40e1eb656233c7011c6f65b7d48658e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85687/","number":"85687","commit":"5e21aecfc057655a022b2ca8c92aea176164156f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85786/","number":"85786","commit":"87126ad8dc4fb7fe67a5541bb7dd38b27b7f38bd"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85823/)
* Commit: 6128b0df8ee838c8b79f389716f74124d575bde3
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85823/execution/node/393/log)

### History
* :broken_heart: [Build #85786](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85786/) failed 87126ad8dc4fb7fe67a5541bb7dd38b27b7f38bd
* :broken_heart: [Build #85687](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85687/) failed 5e21aecfc057655a022b2ca8c92aea176164156f
* :broken_heart: [Build #85632](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85632/) failed 37e3372fa40e1eb656233c7011c6f65b7d48658e
* :broken_heart: [Build #85621](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85621/) failed b7bf3bed23364d4f5059709392cf03cc6d49ce8b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85621/","number":"85621","commit":"b7bf3bed23364d4f5059709392cf03cc6d49ce8b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85632/","number":"85632","commit":"37e3372fa40e1eb656233c7011c6f65b7d48658e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85687/","number":"85687","commit":"5e21aecfc057655a022b2ca8c92aea176164156f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85786/","number":"85786","commit":"87126ad8dc4fb7fe67a5541bb7dd38b27b7f38bd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85823/","number":"85823","commit":"6128b0df8ee838c8b79f389716f74124d575bde3"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85851/)
* Commit: b4005c90199b902305cc0ffa658ed7d46d4a3d74
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85851/execution/node/375/log)

### History
* :broken_heart: [Build #85823](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85823/) failed 6128b0df8ee838c8b79f389716f74124d575bde3
* :broken_heart: [Build #85786](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85786/) failed 87126ad8dc4fb7fe67a5541bb7dd38b27b7f38bd
* :broken_heart: [Build #85687](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85687/) failed 5e21aecfc057655a022b2ca8c92aea176164156f
* :broken_heart: [Build #85632](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85632/) failed 37e3372fa40e1eb656233c7011c6f65b7d48658e
* :broken_heart: [Build #85621](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85621/) failed b7bf3bed23364d4f5059709392cf03cc6d49ce8b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85621/","number":"85621","commit":"b7bf3bed23364d4f5059709392cf03cc6d49ce8b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85632/","number":"85632","commit":"37e3372fa40e1eb656233c7011c6f65b7d48658e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85687/","number":"85687","commit":"5e21aecfc057655a022b2ca8c92aea176164156f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85786/","number":"85786","commit":"87126ad8dc4fb7fe67a5541bb7dd38b27b7f38bd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85823/","number":"85823","commit":"6128b0df8ee838c8b79f389716f74124d575bde3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85851/","number":"85851","commit":"b4005c90199b902305cc0ffa658ed7d46d4a3d74"}]}
PIPELINE-->

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85876/)
* Commit: 2b86ce2b4a6d5104d851f9ea507905b80bdafa43
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85876/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Build X-Pack Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85876/execution/node/125/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85876/execution/node/137/log)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ‼️ ERROR: metrics for 2b86ce2b4a6d5104d851f9ea507905b80bdafa43 were not reported

### History
* :broken_heart: [Build #85851](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85851/) failed b4005c90199b902305cc0ffa658ed7d46d4a3d74
* :broken_heart: [Build #85823](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85823/) failed 6128b0df8ee838c8b79f389716f74124d575bde3
* :broken_heart: [Build #85786](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85786/) failed 87126ad8dc4fb7fe67a5541bb7dd38b27b7f38bd
* :broken_heart: [Build #85687](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85687/) failed 5e21aecfc057655a022b2ca8c92aea176164156f
* :broken_heart: [Build #85632](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85632/) failed 37e3372fa40e1eb656233c7011c6f65b7d48658e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85632/","number":"85632","commit":"37e3372fa40e1eb656233c7011c6f65b7d48658e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85687/","number":"85687","commit":"5e21aecfc057655a022b2ca8c92aea176164156f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85786/","number":"85786","commit":"87126ad8dc4fb7fe67a5541bb7dd38b27b7f38bd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85823/","number":"85823","commit":"6128b0df8ee838c8b79f389716f74124d575bde3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85851/","number":"85851","commit":"b4005c90199b902305cc0ffa658ed7d46d4a3d74"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85876/","number":"85876","commit":"2b86ce2b4a6d5104d851f9ea507905b80bdafa43"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/apm-ui (Team:apm)

elasticmachine commented:
Pinging @elastic/uptime (Team:uptime)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85927/)
* Commit: 5d276670ead5a840b3678b8a8731e8512f7d184f
* This comment will update when the build is complete

### Failed CI Steps
* [X-Pack Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85927/execution/node/339/log)

### History
* :broken_heart: [Build #85851](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85851/) failed b4005c90199b902305cc0ffa658ed7d46d4a3d74
* :broken_heart: [Build #85876](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85876/) failed 2b86ce2b4a6d5104d851f9ea507905b80bdafa43
* :broken_heart: [Build #85823](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85823/) failed 6128b0df8ee838c8b79f389716f74124d575bde3
* :broken_heart: [Build #85786](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85786/) failed 87126ad8dc4fb7fe67a5541bb7dd38b27b7f38bd
* :broken_heart: [Build #85687](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85687/) failed 5e21aecfc057655a022b2ca8c92aea176164156f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85687/","number":"85687","commit":"5e21aecfc057655a022b2ca8c92aea176164156f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85786/","number":"85786","commit":"87126ad8dc4fb7fe67a5541bb7dd38b27b7f38bd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85823/","number":"85823","commit":"6128b0df8ee838c8b79f389716f74124d575bde3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85876/","number":"85876","commit":"2b86ce2b4a6d5104d851f9ea507905b80bdafa43"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85851/","number":"85851","commit":"b4005c90199b902305cc0ffa658ed7d46d4a3d74"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85927/","number":"85927","commit":"5d276670ead5a840b3678b8a8731e8512f7d184f"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85937/)
* Commit: 076cde3e9d9614770380744760eb7073ab06c7ca
* This comment will update when the build is complete

### Failed CI Steps
* [X-Pack Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85937/execution/node/339/log)

### History
* :broken_heart: [Build #85927](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85927/) failed 5d276670ead5a840b3678b8a8731e8512f7d184f
* :broken_heart: [Build #85851](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85851/) failed b4005c90199b902305cc0ffa658ed7d46d4a3d74
* :broken_heart: [Build #85876](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85876/) failed 2b86ce2b4a6d5104d851f9ea507905b80bdafa43
* :broken_heart: [Build #85823](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85823/) failed 6128b0df8ee838c8b79f389716f74124d575bde3
* :broken_heart: [Build #85786](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85786/) failed 87126ad8dc4fb7fe67a5541bb7dd38b27b7f38bd

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85786/","number":"85786","commit":"87126ad8dc4fb7fe67a5541bb7dd38b27b7f38bd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85823/","number":"85823","commit":"6128b0df8ee838c8b79f389716f74124d575bde3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85876/","number":"85876","commit":"2b86ce2b4a6d5104d851f9ea507905b80bdafa43"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85851/","number":"85851","commit":"b4005c90199b902305cc0ffa658ed7d46d4a3d74"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85927/","number":"85927","commit":"5d276670ead5a840b3678b8a8731e8512f7d184f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85937/","number":"85937","commit":"076cde3e9d9614770380744760eb7073ab06c7ca"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85957/)
* Commit: 3f5cedf67fae27f77b24e16ec0a492a8cc0a24fd
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85957/execution/node/371/log)

### History
* :broken_heart: [Build #85937](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85937/) failed 076cde3e9d9614770380744760eb7073ab06c7ca
* :broken_heart: [Build #85927](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85927/) failed 5d276670ead5a840b3678b8a8731e8512f7d184f
* :broken_heart: [Build #85851](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85851/) failed b4005c90199b902305cc0ffa658ed7d46d4a3d74
* :broken_heart: [Build #85876](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85876/) failed 2b86ce2b4a6d5104d851f9ea507905b80bdafa43
* :broken_heart: [Build #85823](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85823/) failed 6128b0df8ee838c8b79f389716f74124d575bde3

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85823/","number":"85823","commit":"6128b0df8ee838c8b79f389716f74124d575bde3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85876/","number":"85876","commit":"2b86ce2b4a6d5104d851f9ea507905b80bdafa43"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85851/","number":"85851","commit":"b4005c90199b902305cc0ffa658ed7d46d4a3d74"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85927/","number":"85927","commit":"5d276670ead5a840b3678b8a8731e8512f7d184f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85937/","number":"85937","commit":"076cde3e9d9614770380744760eb7073ab06c7ca"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85957/","number":"85957","commit":"3f5cedf67fae27f77b24e16ec0a492a8cc0a24fd"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85965/)
* Commit: a631b95b40174b7fe82f32fc9c8ea042a0841575
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup10](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85965/execution/node/677/log)
* [Execute xpack-kibana-ciGroup10](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85965/execution/node/745/log)

### History
* :broken_heart: [Build #85937](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85937/) failed 076cde3e9d9614770380744760eb7073ab06c7ca
* :broken_heart: [Build #85927](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85927/) failed 5d276670ead5a840b3678b8a8731e8512f7d184f
* :broken_heart: [Build #85957](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85957/) failed 3f5cedf67fae27f77b24e16ec0a492a8cc0a24fd
* :broken_heart: [Build #85851](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85851/) failed b4005c90199b902305cc0ffa658ed7d46d4a3d74
* :broken_heart: [Build #85876](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85876/) failed 2b86ce2b4a6d5104d851f9ea507905b80bdafa43

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85876/","number":"85876","commit":"2b86ce2b4a6d5104d851f9ea507905b80bdafa43"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85851/","number":"85851","commit":"b4005c90199b902305cc0ffa658ed7d46d4a3d74"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85957/","number":"85957","commit":"3f5cedf67fae27f77b24e16ec0a492a8cc0a24fd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85927/","number":"85927","commit":"5d276670ead5a840b3678b8a8731e8512f7d184f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85937/","number":"85937","commit":"076cde3e9d9614770380744760eb7073ab06c7ca"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85965/","number":"85965","commit":"a631b95b40174b7fe82f32fc9c8ea042a0841575"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86137/)
* Commit: d398759bdbdfb71423d9d74c911cc24e2a6b59eb
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup10](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86137/execution/node/677/log)
* [Execute xpack-kibana-ciGroup10](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86137/execution/node/745/log)

### History
* :broken_heart: [Build #85965](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85965/) failed a631b95b40174b7fe82f32fc9c8ea042a0841575
* :broken_heart: [Build #85937](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85937/) failed 076cde3e9d9614770380744760eb7073ab06c7ca
* :broken_heart: [Build #85927](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85927/) failed 5d276670ead5a840b3678b8a8731e8512f7d184f
* :broken_heart: [Build #85957](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85957/) failed 3f5cedf67fae27f77b24e16ec0a492a8cc0a24fd
* :broken_heart: [Build #85851](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85851/) failed b4005c90199b902305cc0ffa658ed7d46d4a3d74

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85851/","number":"85851","commit":"b4005c90199b902305cc0ffa658ed7d46d4a3d74"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85957/","number":"85957","commit":"3f5cedf67fae27f77b24e16ec0a492a8cc0a24fd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85927/","number":"85927","commit":"5d276670ead5a840b3678b8a8731e8512f7d184f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85937/","number":"85937","commit":"076cde3e9d9614770380744760eb7073ab06c7ca"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85965/","number":"85965","commit":"a631b95b40174b7fe82f32fc9c8ea042a0841575"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86137/","number":"86137","commit":"d398759bdbdfb71423d9d74c911cc24e2a6b59eb"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86155/)
* Commit: 77522955a8d2b419dd73a1fda627a5e8de29b62a

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### @kbn/optimizer bundle module count
| id | [before](https://github.com/elastic/kibana/commit/3c99839cf7a624e7a8201f8636ae8c9b10cd8a9c) | [after](https://github.com/elastic/kibana/commit/77522955a8d2b419dd73a1fda627a5e8de29b62a) | diff |
| --- | --- | --- | --- |
| `maps` | 620 | 623 | +3 |

#### async chunks size
| id | [before](https://github.com/elastic/kibana/commit/3c99839cf7a624e7a8201f8636ae8c9b10cd8a9c) | [after](https://github.com/elastic/kibana/commit/77522955a8d2b419dd73a1fda627a5e8de29b62a) | diff |
| --- | --- | --- | --- |
| `apm` | 3.2MB | 3.2MB | +22.0B |
| `maps` | 2.8MB | 2.7MB | -6.8KB |
| `securitySolution` | 7.8MB | 7.8MB | -32.0B |
| `uptime` | 1.1MB | 1.1MB | -34.0B |
| total |  |  | -6.8KB |

#### page load bundle size
| id | [before](https://github.com/elastic/kibana/commit/3c99839cf7a624e7a8201f8636ae8c9b10cd8a9c) | [after](https://github.com/elastic/kibana/commit/77522955a8d2b419dd73a1fda627a5e8de29b62a) | diff |
| --- | --- | --- | --- |
| `canvas` | 908.8KB | 908.8KB | +22.0B |
| `maps` | 153.9KB | 149.0KB | -4.9KB |
| total |  |  | -4.9KB |

### History
* :broken_heart: [Build #86137](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86137/) failed d398759bdbdfb71423d9d74c911cc24e2a6b59eb
* :broken_heart: [Build #85965](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85965/) failed a631b95b40174b7fe82f32fc9c8ea042a0841575
* :broken_heart: [Build #85937](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85937/) failed 076cde3e9d9614770380744760eb7073ab06c7ca
* :broken_heart: [Build #85927](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85927/) failed 5d276670ead5a840b3678b8a8731e8512f7d184f
* :broken_heart: [Build #85957](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85957/) failed 3f5cedf67fae27f77b24e16ec0a492a8cc0a24fd

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85957/","number":"85957","commit":"3f5cedf67fae27f77b24e16ec0a492a8cc0a24fd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85927/","number":"85927","commit":"5d276670ead5a840b3678b8a8731e8512f7d184f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85937/","number":"85937","commit":"076cde3e9d9614770380744760eb7073ab06c7ca"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85965/","number":"85965","commit":"a631b95b40174b7fe82f32fc9c8ea042a0841575"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86137/","number":"86137","commit":"d398759bdbdfb71423d9d74c911cc24e2a6b59eb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86155/","number":"86155","commit":"77522955a8d2b419dd73a1fda627a5e8de29b62a"}]}
PIPELINE-->

ThomThomson reviewed (commented):
Tested locally in chrome.

Code LGTM. Everything looks super clean.

Adding to library, cloning panel, and unlinking all work flawlessly with by reference and by value panels!

One issue I found is that editing 'by reference' maps from a dashboard then hitting  'save and return' doesn't apply the changes properly. You need to press 'save to maps' to see your changes. Not totally sure what's going on there.

I also have some tiny UX nits that could maybe be addressed in a followup:

- When editing a 'by reference' map from a dashboard, the button to the left of 'save and return' still says 'save to maps'. If you're editing though that means it's already been saved to maps. In Lens, in that case, the button reads 'save as'. 

- When editing a 'by reference' panel from dashboard I would say that it makes sense to have the 'maps' breadcrumb also present - so it would be `Dashboard / Maps / My cool Map` when editing from dashboard in by reference mode.

- In both visualize and lens, new by value panels are saved with no title at all. Editing them shows something like `Dashboard / Edit visualization` in the breadcrumbs, and when you're in view mode on the dashboard, those blank titles just aren't shown. In Maps so far, by value maps are all named **New Map**, which means the users will have customize each panel's title, or hide the panel titles - else they will potentially have a bunch of 'New Map' on their dashboard.

Additionally, while testing I noticed that saving a map with filters applied doesn't actually apply when they are on a dashboard. This behaviour was the same in maps in 7.10 BC 4, and in 7.9.2 but I'm not sure if it's intentional or not. If not, maybe a separate issue.

Overall, this is a huge change, and will contribute _a lot_ to the seamless feeling we hope to capture when we release Time to Visualize!

nreese commented:
> Additionally, while testing I noticed that saving a map with filters applied doesn't actually apply when they are on a dashboard. This behaviour was the same in maps in 7.10 BC 4, and in 7.9.2 but I'm not sure if it's intentional or not. If not, maybe a separate issue.

This is as expected behavior. Query, filters, and time saved with a map are not transferred to Dashboard

thomasneirynck commented on x-pack/plugins/maps/public/lazy_load_bundle/index.ts:
```diff
@@ -4,50 +4,32 @@
  * you may not use this file except in compliance with the Elastic License.
```
wow..

thomasneirynck reviewed (commented):
this is such a fun PR to play with. kibana and maps feel twice as fast.

the simplifcation is impressive.

Any ideas on how to test? It'd make sense to add some functional test for the new flow.

ThomThomson commented:
@thomasneirynck - great timing with the question of adding functional tests. This PR https://github.com/elastic/kibana/pull/82441/files was recently merged which should allow us to start writing tests with `dashboard.allowByValueEmbeddables` on.

nreese commented on x-pack/plugins/maps/public/routes/map_app/saved_map/saved_map.ts:
```diff
@@ -0,0 +1,333 @@
+    return this._attributes;
+  }
+
+  hasUnsavedChanges = () => {
```
SavedMap.hasUnsavedChanges is also called from MapApp onAppLeave so lamba function is needed there as well.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86549/)
* Commit: afd537591e3a54fa4680ba231fba144f692a83eb

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### @kbn/optimizer bundle module count
| id | [before](https://github.com/elastic/kibana/commit/ae20a3a2a94a131dc6aafa84a22b4508e9ba1b9b) | [after](https://github.com/elastic/kibana/commit/afd537591e3a54fa4680ba231fba144f692a83eb) | diff |
| --- | --- | --- | --- |
| `maps` | 620 | 623 | +3 |

#### async chunks size
| id | [before](https://github.com/elastic/kibana/commit/ae20a3a2a94a131dc6aafa84a22b4508e9ba1b9b) | [after](https://github.com/elastic/kibana/commit/afd537591e3a54fa4680ba231fba144f692a83eb) | diff |
| --- | --- | --- | --- |
| `apm` | 3.2MB | 3.2MB | +22.0B |
| `maps` | 2.8MB | 2.7MB | -6.8KB |
| `securitySolution` | 7.8MB | 7.8MB | -32.0B |
| `uptime` | 1.1MB | 1.1MB | -34.0B |
| total |  |  | -6.8KB |

#### page load bundle size
| id | [before](https://github.com/elastic/kibana/commit/ae20a3a2a94a131dc6aafa84a22b4508e9ba1b9b) | [after](https://github.com/elastic/kibana/commit/afd537591e3a54fa4680ba231fba144f692a83eb) | diff |
| --- | --- | --- | --- |
| `canvas` | 908.9KB | 908.9KB | +22.0B |
| `maps` | 153.9KB | 149.0KB | -4.9KB |
| total |  |  | -4.9KB |

### History
* :green_heart: [Build #86155](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86155/) succeeded 77522955a8d2b419dd73a1fda627a5e8de29b62a
* :broken_heart: [Build #86137](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86137/) failed d398759bdbdfb71423d9d74c911cc24e2a6b59eb
* :broken_heart: [Build #85965](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85965/) failed a631b95b40174b7fe82f32fc9c8ea042a0841575
* :broken_heart: [Build #85937](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85937/) failed 076cde3e9d9614770380744760eb7073ab06c7ca
* :broken_heart: [Build #85927](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85927/) failed 5d276670ead5a840b3678b8a8731e8512f7d184f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85927/","number":"85927","commit":"5d276670ead5a840b3678b8a8731e8512f7d184f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85937/","number":"85937","commit":"076cde3e9d9614770380744760eb7073ab06c7ca"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85965/","number":"85965","commit":"a631b95b40174b7fe82f32fc9c8ea042a0841575"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86137/","number":"86137","commit":"d398759bdbdfb71423d9d74c911cc24e2a6b59eb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86155/","number":"86155","commit":"77522955a8d2b419dd73a1fda627a5e8de29b62a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86549/","number":"86549","commit":"afd537591e3a54fa4680ba231fba144f692a83eb"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86557/)
* Commit: c13676f80e0e234ed9e7ae94857d2a7c4fa1951b

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### @kbn/optimizer bundle module count
| id | [before](https://github.com/elastic/kibana/commit/ae20a3a2a94a131dc6aafa84a22b4508e9ba1b9b) | [after](https://github.com/elastic/kibana/commit/c13676f80e0e234ed9e7ae94857d2a7c4fa1951b) | diff |
| --- | --- | --- | --- |
| `maps` | 620 | 623 | +3 |

#### async chunks size
| id | [before](https://github.com/elastic/kibana/commit/ae20a3a2a94a131dc6aafa84a22b4508e9ba1b9b) | [after](https://github.com/elastic/kibana/commit/c13676f80e0e234ed9e7ae94857d2a7c4fa1951b) | diff |
| --- | --- | --- | --- |
| `apm` | 3.2MB | 3.2MB | +22.0B |
| `maps` | 2.8MB | 2.7MB | -6.9KB |
| `securitySolution` | 7.8MB | 7.8MB | -32.0B |
| `uptime` | 1.1MB | 1.1MB | -34.0B |
| total |  |  | -6.9KB |

#### page load bundle size
| id | [before](https://github.com/elastic/kibana/commit/ae20a3a2a94a131dc6aafa84a22b4508e9ba1b9b) | [after](https://github.com/elastic/kibana/commit/c13676f80e0e234ed9e7ae94857d2a7c4fa1951b) | diff |
| --- | --- | --- | --- |
| `canvas` | 908.9KB | 908.9KB | +22.0B |
| `maps` | 153.9KB | 149.0KB | -4.9KB |
| total |  |  | -4.9KB |

### History
* :green_heart: [Build #86549](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86549/) succeeded afd537591e3a54fa4680ba231fba144f692a83eb
* :green_heart: [Build #86155](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86155/) succeeded 77522955a8d2b419dd73a1fda627a5e8de29b62a
* :broken_heart: [Build #86137](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86137/) failed d398759bdbdfb71423d9d74c911cc24e2a6b59eb
* :broken_heart: [Build #85965](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85965/) failed a631b95b40174b7fe82f32fc9c8ea042a0841575
* :broken_heart: [Build #85937](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85937/) failed 076cde3e9d9614770380744760eb7073ab06c7ca

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85937/","number":"85937","commit":"076cde3e9d9614770380744760eb7073ab06c7ca"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85965/","number":"85965","commit":"a631b95b40174b7fe82f32fc9c8ea042a0841575"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86137/","number":"86137","commit":"d398759bdbdfb71423d9d74c911cc24e2a6b59eb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86155/","number":"86155","commit":"77522955a8d2b419dd73a1fda627a5e8de29b62a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86549/","number":"86549","commit":"afd537591e3a54fa4680ba231fba144f692a83eb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86557/","number":"86557","commit":"c13676f80e0e234ed9e7ae94857d2a7c4fa1951b"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86561/)
* Commit: ace3e697e4bbe8d025905b96a26db99db2a29508

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### @kbn/optimizer bundle module count
| id | [before](https://github.com/elastic/kibana/commit/ae20a3a2a94a131dc6aafa84a22b4508e9ba1b9b) | [after](https://github.com/elastic/kibana/commit/ace3e697e4bbe8d025905b96a26db99db2a29508) | diff |
| --- | --- | --- | --- |
| `maps` | 620 | 623 | +3 |

#### async chunks size
| id | [before](https://github.com/elastic/kibana/commit/ae20a3a2a94a131dc6aafa84a22b4508e9ba1b9b) | [after](https://github.com/elastic/kibana/commit/ace3e697e4bbe8d025905b96a26db99db2a29508) | diff |
| --- | --- | --- | --- |
| `apm` | 3.2MB | 3.2MB | +22.0B |
| `maps` | 2.8MB | 2.7MB | -6.9KB |
| `securitySolution` | 7.8MB | 7.8MB | -32.0B |
| `uptime` | 1.1MB | 1.1MB | -34.0B |
| total |  |  | -6.9KB |

#### page load bundle size
| id | [before](https://github.com/elastic/kibana/commit/ae20a3a2a94a131dc6aafa84a22b4508e9ba1b9b) | [after](https://github.com/elastic/kibana/commit/ace3e697e4bbe8d025905b96a26db99db2a29508) | diff |
| --- | --- | --- | --- |
| `canvas` | 908.9KB | 908.9KB | +22.0B |
| `maps` | 153.9KB | 149.0KB | -4.9KB |
| total |  |  | -4.9KB |

### History
* :green_heart: [Build #86557](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86557/) succeeded c13676f80e0e234ed9e7ae94857d2a7c4fa1951b
* :green_heart: [Build #86549](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86549/) succeeded afd537591e3a54fa4680ba231fba144f692a83eb
* :green_heart: [Build #86155](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86155/) succeeded 77522955a8d2b419dd73a1fda627a5e8de29b62a
* :broken_heart: [Build #86137](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86137/) failed d398759bdbdfb71423d9d74c911cc24e2a6b59eb
* :broken_heart: [Build #85965](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85965/) failed a631b95b40174b7fe82f32fc9c8ea042a0841575

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85965/","number":"85965","commit":"a631b95b40174b7fe82f32fc9c8ea042a0841575"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86137/","number":"86137","commit":"d398759bdbdfb71423d9d74c911cc24e2a6b59eb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86155/","number":"86155","commit":"77522955a8d2b419dd73a1fda627a5e8de29b62a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86549/","number":"86549","commit":"afd537591e3a54fa4680ba231fba144f692a83eb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86557/","number":"86557","commit":"c13676f80e0e234ed9e7ae94857d2a7c4fa1951b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86561/","number":"86561","commit":"ace3e697e4bbe8d025905b96a26db99db2a29508"}]}
PIPELINE-->

nreese commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86768/)
* Commit: 92feba62eb3e3e837cb279d6469061b99e0d978f

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### @kbn/optimizer bundle module count
| id | [before](https://github.com/elastic/kibana/commit/97e2dc853b5d0d12b87691bb447f6475de0bbfec) | [after](https://github.com/elastic/kibana/commit/92feba62eb3e3e837cb279d6469061b99e0d978f) | diff |
| --- | --- | --- | --- |
| `maps` | 620 | 623 | +3 |

#### async chunks size
| id | [before](https://github.com/elastic/kibana/commit/97e2dc853b5d0d12b87691bb447f6475de0bbfec) | [after](https://github.com/elastic/kibana/commit/92feba62eb3e3e837cb279d6469061b99e0d978f) | diff |
| --- | --- | --- | --- |
| `apm` | 3.2MB | 3.2MB | +22.0B |
| `maps` | 2.8MB | 2.7MB | -6.9KB |
| `securitySolution` | 7.8MB | 7.8MB | -32.0B |
| `uptime` | 1.1MB | 1.1MB | -34.0B |
| total |  |  | -6.9KB |

#### page load bundle size
| id | [before](https://github.com/elastic/kibana/commit/97e2dc853b5d0d12b87691bb447f6475de0bbfec) | [after](https://github.com/elastic/kibana/commit/92feba62eb3e3e837cb279d6469061b99e0d978f) | diff |
| --- | --- | --- | --- |
| `canvas` | 908.9KB | 908.9KB | +22.0B |
| `maps` | 153.9KB | 149.0KB | -4.9KB |
| total |  |  | -4.9KB |

### History
* :green_heart: [Build #86561](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86561/) succeeded ace3e697e4bbe8d025905b96a26db99db2a29508
* :green_heart: [Build #86557](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86557/) succeeded c13676f80e0e234ed9e7ae94857d2a7c4fa1951b
* :green_heart: [Build #86549](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86549/) succeeded afd537591e3a54fa4680ba231fba144f692a83eb
* :green_heart: [Build #86155](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86155/) succeeded 77522955a8d2b419dd73a1fda627a5e8de29b62a
* :broken_heart: [Build #86137](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86137/) failed d398759bdbdfb71423d9d74c911cc24e2a6b59eb

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86137/","number":"86137","commit":"d398759bdbdfb71423d9d74c911cc24e2a6b59eb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86155/","number":"86155","commit":"77522955a8d2b419dd73a1fda627a5e8de29b62a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86549/","number":"86549","commit":"afd537591e3a54fa4680ba231fba144f692a83eb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86557/","number":"86557","commit":"c13676f80e0e234ed9e7ae94857d2a7c4fa1951b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86561/","number":"86561","commit":"ace3e697e4bbe8d025905b96a26db99db2a29508"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86768/","number":"86768","commit":"92feba62eb3e3e837cb279d6469061b99e0d978f"}]}
PIPELINE-->

nreese commented:
@ThomThomson Thanks for all the great feedback on the little stuff like breadcrumbs, button labels, and page titles. These are great things to make consistent with Lens to provide a cohesive user experience. I have addressed the following.

> When editing a 'by reference' map from a dashboard, the button to the left of 'save and return' still says 'save to maps'. If you're editing though that means it's already been saved to maps. In Lens, in that case, the button reads 'save as'.

Resolved. Added proper logic to set button label when "by value" or "by reference"

> When editing a 'by reference' panel from dashboard I would say that it makes sense to have the 'maps' breadcrumb also present - so it would be Dashboard / Maps / My cool Map when editing from dashboard in by reference mode.

Resolved. Added proper logic to include "Maps" in path when "by reference"

In both visualize and lens, new by value panels are saved with no title at all. Editing them shows something like Dashboard / Edit visualization in the breadcrumbs, and when you're in view mode on the dashboard, those blank titles just aren't shown. In Maps so far, by value maps are all named New Map, which means the users will have customize each panel's title, or hide the panel titles - else they will potentially have a bunch of 'New Map' on their dashboard.

Resolved. Map title now defaults to empty string like lens. When you create a new map either from Maps or Dashboard, the page title is "Create". When you edit a map "by value", the page title is "Edit map". When you edit a map "by reference", the page title is the saved object title.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86820/)
* Commit: 7921138b0366580da143c9396aa20e27278b24e1
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86820/execution/node/375/log)

### History
* :green_heart: [Build #86768](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86768/) succeeded 92feba62eb3e3e837cb279d6469061b99e0d978f
* :green_heart: [Build #86561](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86561/) succeeded ace3e697e4bbe8d025905b96a26db99db2a29508
* :green_heart: [Build #86557](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86557/) succeeded c13676f80e0e234ed9e7ae94857d2a7c4fa1951b
* :green_heart: [Build #86549](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86549/) succeeded afd537591e3a54fa4680ba231fba144f692a83eb
* :green_heart: [Build #86155](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86155/) succeeded 77522955a8d2b419dd73a1fda627a5e8de29b62a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86155/","number":"86155","commit":"77522955a8d2b419dd73a1fda627a5e8de29b62a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86549/","number":"86549","commit":"afd537591e3a54fa4680ba231fba144f692a83eb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86557/","number":"86557","commit":"c13676f80e0e234ed9e7ae94857d2a7c4fa1951b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86561/","number":"86561","commit":"ace3e697e4bbe8d025905b96a26db99db2a29508"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86768/","number":"86768","commit":"92feba62eb3e3e837cb279d6469061b99e0d978f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86820/","number":"86820","commit":"7921138b0366580da143c9396aa20e27278b24e1"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86844/)
* Commit: a5f4853d77d5259271306eac94efcfa5f482ffac

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/97e2dc853b5d0d12b87691bb447f6475de0bbfec) | [after](https://github.com/elastic/kibana/commit/a5f4853d77d5259271306eac94efcfa5f482ffac) | diff |
| --- | --- | --- | --- |
| `maps` | 620 | 623 | +3 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/97e2dc853b5d0d12b87691bb447f6475de0bbfec) | [after](https://github.com/elastic/kibana/commit/a5f4853d77d5259271306eac94efcfa5f482ffac) | diff |
| --- | --- | --- | --- |
| `apm` | 3.2MB | 3.2MB | +22.0B |
| `maps` | 2.8MB | 2.7MB | -6.2KB |
| `securitySolution` | 7.8MB | 7.8MB | -32.0B |
| `uptime` | 1.1MB | 1.1MB | -34.0B |
| total |  |  | -6.3KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/97e2dc853b5d0d12b87691bb447f6475de0bbfec) | [after](https://github.com/elastic/kibana/commit/a5f4853d77d5259271306eac94efcfa5f482ffac) | diff |
| --- | --- | --- | --- |
| `canvas` | 908.9KB | 908.9KB | +22.0B |
| `maps` | 153.9KB | 149.0KB | -4.9KB |
| total |  |  | -4.9KB |

### History
* :broken_heart: [Build #86820](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86820/) failed 7921138b0366580da143c9396aa20e27278b24e1
* :green_heart: [Build #86768](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86768/) succeeded 92feba62eb3e3e837cb279d6469061b99e0d978f
* :green_heart: [Build #86561](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86561/) succeeded ace3e697e4bbe8d025905b96a26db99db2a29508
* :green_heart: [Build #86557](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86557/) succeeded c13676f80e0e234ed9e7ae94857d2a7c4fa1951b
* :green_heart: [Build #86549](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86549/) succeeded afd537591e3a54fa4680ba231fba144f692a83eb

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86549/","number":"86549","commit":"afd537591e3a54fa4680ba231fba144f692a83eb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86557/","number":"86557","commit":"c13676f80e0e234ed9e7ae94857d2a7c4fa1951b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86561/","number":"86561","commit":"ace3e697e4bbe8d025905b96a26db99db2a29508"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86768/","number":"86768","commit":"92feba62eb3e3e837cb279d6469061b99e0d978f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86820/","number":"86820","commit":"7921138b0366580da143c9396aa20e27278b24e1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86844/","number":"86844","commit":"a5f4853d77d5259271306eac94efcfa5f482ffac"}]}
PIPELINE-->

thomasneirynck reviewed (approved):
lgtm. So great, this flow feels really fast. Thanks for simplifying how the maps is bootstrapped in the page of the Maps-app and inside the embeddable.

ThomThomson reviewed (approved):
Tested again on Chrome. Everything works great! Thanks again for looking into those consistency nits.

LGTM

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/87151/)
* Commit: 45c255371f9ab895a00c3098d407595bb019d175

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/28305237a318212715491c4655f2b26e0f3d77bc) | [after](https://github.com/elastic/kibana/commit/45c255371f9ab895a00c3098d407595bb019d175) | diff |
| --- | --- | --- | --- |
| `maps` | 624 | 627 | +3 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/28305237a318212715491c4655f2b26e0f3d77bc) | [after](https://github.com/elastic/kibana/commit/45c255371f9ab895a00c3098d407595bb019d175) | diff |
| --- | --- | --- | --- |
| `apm` | 3.1MB | 3.1MB | +22.0B |
| `maps` | 2.8MB | 2.8MB | -6.2KB |
| `securitySolution` | 7.8MB | 7.8MB | -32.0B |
| `uptime` | 983.1KB | 983.1KB | -34.0B |
| total |  |  | -6.3KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/28305237a318212715491c4655f2b26e0f3d77bc) | [after](https://github.com/elastic/kibana/commit/45c255371f9ab895a00c3098d407595bb019d175) | diff |
| --- | --- | --- | --- |
| `canvas` | 908.9KB | 908.9KB | +22.0B |
| `maps` | 155.6KB | 150.7KB | -4.9KB |
| total |  |  | -4.9KB |

### History
* :green_heart: [Build #86844](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86844/) succeeded a5f4853d77d5259271306eac94efcfa5f482ffac
* :broken_heart: [Build #86820](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86820/) failed 7921138b0366580da143c9396aa20e27278b24e1
* :green_heart: [Build #86768](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86768/) succeeded 92feba62eb3e3e837cb279d6469061b99e0d978f
* :green_heart: [Build #86561](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86561/) succeeded ace3e697e4bbe8d025905b96a26db99db2a29508
* :green_heart: [Build #86557](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86557/) succeeded c13676f80e0e234ed9e7ae94857d2a7c4fa1951b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86557/","number":"86557","commit":"c13676f80e0e234ed9e7ae94857d2a7c4fa1951b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86561/","number":"86561","commit":"ace3e697e4bbe8d025905b96a26db99db2a29508"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86768/","number":"86768","commit":"92feba62eb3e3e837cb279d6469061b99e0d978f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86820/","number":"86820","commit":"7921138b0366580da143c9396aa20e27278b24e1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86844/","number":"86844","commit":"a5f4853d77d5259271306eac94efcfa5f482ffac"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/87151/","number":"87151","commit":"45c255371f9ab895a00c3098d407595bb019d175"}]}
PIPELINE-->
